containers/haskell/.devcontainer/Dockerfile

```diff
@@ -20,6 +38,22 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
     && /bin/bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true"\
     && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
 
-# [Optional] Uncomment this section to install additional OS packages.
-# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-#     && apt-get -y install --no-install-recommends <your-package-list-here>
+# Install latest GHCup in the non-root user home
+USER $USERNAME
+
+RUN mkdir -p "$HOME/.ghcup/bin"
+RUN curl -LJ "https://downloads.haskell.org/~ghcup/x86_64-linux-ghcup" -o "$HOME/.ghcup/bin/ghcup"
+RUN chmod +x "$HOME/.ghcup/bin/ghcup"
+ENV PATH="/home/$USERNAME/.cabal/bin:/home/$USERNAME/.ghcup/bin:$PATH"
+
+# [Choice] GHC version: recommended, latest, 9.2, 9.0, 8.10, 8.8, 8.6
+ARG GHC_VERSION="recommended"
+
+# Use GHCup to install versions of main utilities
+# If you prefer to let the Haskell extension install everything on demand,
+# comment out the lines below. In that case, you may need to manually run `cabal update`.
+RUN ghcup install ghc "${GHC_VERSION}" --set
+RUN ghcup install cabal recommended --set
+RUN ghcup install stack recommended --set
+RUN ghcup install hls recommended --set
+RUN cabal update
```

Comment:
It would be better to reduce the number of layers. Please check the README.
https://github.com/microsoft/vscode-dev-containers#why-do-dockerfiles-in-this-repo-use-run-statements-with-commands-separated-by-